### PR TITLE
Replace all embeds with Vals and remove references to `set`

### PR DESCRIPTION
**Pull request by Xkonti**
Closes https://github.com/val-town/val-town-docs/issues/57
Closes https://github.com/val-town/val-town-docs/issues/58

## Done
- Replaced all embedded vals with code blocks
- Replaced all found references to `std/set` with `std/blob`
- Replaced all Environment Variables referenced via `process.env` with `Deno.env.get()`
- Replaced some remnants of "secrets" with "environment variables"
- Replaced some remaining `@user` imports with proper JS imports

## Omitted

- Types:
  - HTTP / Basic examples - embeds are used to display the results of API requests
- Guides:
  - Push Notifications - displays user vals
  - QR Code - displays user vals
  - Saving data from a web page - needs rewrite
  - Weather - displays user vals
  - Website Uptime Tracker - displays user vals
- Integrations:
  - Discord welcome bot - needs more rework
  - SQLite wasm - needs more rework to utilize blob storage

### CODE REVIEW

**Comment by stevekrouse** [comment]
This is awesome! Thanks so much @Xkonti ❤️

**Comment by Xkonti** [comment]
Nearly all the docs updated with exception of those mentioned in the description. I think it would be good to create PRs for:
- Guides / Saving data from a web page
- Integrations / Discord welcome bot
- Integrations / SQLite wasm

File: src/content/docs/api/aliases.mdx [hunk 1]
```diff
@@ -15,8 +13,23 @@ import Val from "@components/Val.astro";
...
 
-<Val url="https://www.val.town/embed/neverstew.aliasValExample" />
+```ts val
+import { alias } from "https://esm.town/v/neverstew/alias";
+
+export const aliasValExample = alias({
+  username: "stevekrouse",
+  valName: "parentReference",
+  // for private vals, pass your val town api token
+  // token: @me.secrets.valtownToken,
```
Comment (stevekrouse) [inline src/content/docs/api/aliases.mdx:33]:
```suggestion
  // for private vals, pass your val town api token
  // token: Deno.env.get('valtown')
```

File: src/content/docs/api/eval.mdx [hunk 2]
```diff
@@ -29,14 +27,34 @@ You can compute with references to public vals, like
 [`https://api.val.town/v1/eval/@stevekrouse.example1+1`](https://api.val.town/v1/eval/@stevekrouse.example1+1).
 
 You can even call vals:
-[`https://api.val.town/v1/eval/@stevekrouse.add(@stevekrouse.example1, @stevekrouse.example2)`](https://api.val.town/v1/eval/@stevekrouse.add(@stevekrouse.example1,%20@stevekrouse.example2).
+[`https://api.val.town/v1/eval/@stevekrouse.add(@stevekrouse.example1, @stevekrouse.example2)`](https://api.val.town/v1/eval/@stevekrouse.add(@stevekrouse.example1,%20@stevekrouse.example2)).
```
Comment (stevekrouse) [inline src/content/docs/api/eval.mdx:30]: This `@` syntax is no longer supported so this doesn't work anymore. Probably easiest to remove this line.

File: src/content/docs/api/my-resources.mdx [hunk 3]
```diff
@@ -18,7 +16,15 @@ This is the same as `/v1/users/{your_user_id}`
 
 :::
 
-<Val url="https://www.val.town/embed/neverstew.getMe" />
+```ts val
+import { fetchJSON } from "https://esm.town/v/stevekrouse/fetchJSON?v=41";
+
+export let getMe = fetchJSON(`https://api.val.town/v1/me`, {
+  headers: {
+    Authorization: `Bearer ${Deno.env.get("valtownToken")}`,
```
Comment (stevekrouse) [inline src/content/docs/api/my-resources.mdx:24]:
```suggestion
    Authorization: `Bearer ${Deno.env.get("valtown")}`,
```

File: src/content/docs/api/vals.mdx [hunk 4]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
 The vals endpoints allow you to manipulate vals.
 
 ## POST `/v1/vals`
 
-<Val url="https://www.val.town/embed/neverstew.postVals" />
+```ts val
+import { createVal } from "https://esm.town/v/neverstew/createVal";
+
+export const postVals = createVal({
+  token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:14]:
```suggestion
  token: Deno.env.get("valtown"),
```

File: src/content/docs/api/vals.mdx [hunk 5]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
+```
 
 ## DELETE `/v1/vals/{val_id}`
 
-<Val url="https://www.val.town/embed/neverstew.deleteValExample" />
+```ts val
+import { deleteVal } from "https://esm.town/v/neverstew/deleteVal";
+
+export const deleteValExample = deleteVal({
+  token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:35]:
```suggestion
  token: Deno.env.get("valtown"),
```

File: src/content/docs/api/vals.mdx [hunk 6]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
+```
 
 ## GET `/v1/vals/{val_id}/versions`
 
-<Val url="https://www.val.town/embed/neverstew.getValVersions" />
+```ts val
+import { valVersions } from "https://esm.town/v/neverstew/valVersions";
+
+export const getValVersions = valVersions({
+  token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:46]:
```suggestion
  token: Deno.env.get("valtown"),
```

File: src/content/docs/api/vals.mdx [hunk 7]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
+
+export let createValVersionExample = (() =>
+  val2({
+    token: Deno.env.get("valtownToken"),
+    id: "543ae134-636f-43a7-bd7f-f766a3d52b47",
+  })
+    .then((val) => ({ id: val.id, code: val.code }))
+    .then(({ id, code }) =>
+      createValVersion({
+        token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:65]:
```suggestion
        token: Deno.env.get("valtown"),
```

File: src/content/docs/api/vals.mdx [hunk 8]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
+```
 
 ## GET `/v1/vals/{val_id}/versions/{version}`
 
-<Val url="https://www.val.town/embed/neverstew.getValVersion" />
+```ts val
+import { valVersion } from "https://esm.town/v/neverstew/valVersion";
+
+export const getValVersion = valVersion({
+  token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:79]:
```suggestion
  token: Deno.env.get("valtown"),
```

File: src/content/docs/api/vals.mdx [hunk 9]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
+```
 
 ## DELETE `/v1/vals/{val_id}/versions/{version}`
 
-<Val url="https://www.val.town/embed/neverstew.deleteValVersionExample" />
+```ts val
+import { deleteValVersion } from "https://esm.town/v/neverstew/deleteValVersion";
+
+export let deleteValVersionExample = deleteValVersion({
+  token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:91]:
```suggestion
  token: Deno.env.get("valtown"),
```

File: src/content/docs/api/vals.mdx [hunk 10]
```diff
@@ -3,38 +3,106 @@ title: Vals
...
+```
 
 ## GET `/v1/vals/{val_id}/runs`
 
-<Val url="https://www.val.town/embed/neverstew.valRunsExample" />
+```ts val
+import { valRuns } from "https://esm.town/v/neverstew/valRuns";
+
+export const valRunsExample = valRuns({
+  token: Deno.env.get("valtownToken"),
```
Comment (stevekrouse) [inline src/content/docs/api/vals.mdx:103]:
```suggestion
  token: Deno.env.get("valtown"),
```

File: src/content/docs/guides/hosted-puppeteer-browserless.mdx [hunk 11]
```diff
@@ -34,10 +32,33 @@ and save it as a Val Town environment variable as `browserlessKey`.
...
-For example, here’s how you scrape the introduction paragraph of OpenAI’s
+For example, here's how you scrape the introduction paragraph of OpenAI's
 wikipedia page.
 
-<Val url="https://www.val.town/embed/vtdocs.browserlessScrapeExample" />
+```ts val
+import { fetchJSON } from "https://esm.town/v/stevekrouse/fetchJSON?v=41";
+
+const res = await fetchJSON(
+  `https://chrome.browserless.io/scrape?token=${Deno.env.get("browserlessKey")}`,
```
Comment (stevekrouse) [inline src/content/docs/guides/hosted-puppeteer-browserless.mdx:42]:
```suggestion
  `https://chrome.browserless.io/scrape?token=${Deno.env.get("browserless")}`,
```

File: src/content/docs/guides/hosted-puppeteer-browserless.mdx [hunk 12]
```diff
@@ -47,30 +68,26 @@ taking screenshots and PDFs of websites.
...
-  return intro;
-})();
+const { PuppeteerDeno } = await import(
+  "https://deno.land/x/puppeteer@16.2.0/src/deno/Puppeteer.ts"
+);
+const puppeteer = new PuppeteerDeno({
+  productName: "chrome",
+});
+const browser = await puppeteer.connect({
+  browserWSEndpoint: `wss://chrome.browserless.io?token=${Deno.env.get("browserlessKey")}`,
```
Comment (stevekrouse) [inline src/content/docs/guides/hosted-puppeteer-browserless.mdx:82]:
```suggestion
  browserWSEndpoint: `wss://chrome.browserless.io?token=${Deno.env.get("browserless")}`,
```

File: src/content/docs/guides/save-html-form-data.mdx [hunk 13]
```diff
@@ -71,9 +96,56 @@ function) gets sent a
...
+    return new Response(
+      `
+<!DOCTYPE html>
+<html>
+<head>
+  <title>Email Form</title>
+</head>
+<body>
+  <!-- Change the action here to THIS val's Express endpoint! -->
+  <form action="https://user-renderFormAndSaveData.web.val.run" method="post">
```
Comment (stevekrouse) [inline src/content/docs/guides/save-html-form-data.mdx:115]: Import and use this function: https://www.val.town/v/stevekrouse/thisWebURL

```suggestion
  <form action="${thisWebURL()}" method="post">
```

File: src/content/docs/guides/web-scraping.mdx [hunk 14]
```diff
@@ -49,4 +47,20 @@ structure. We also recommend
...
+
+const html = await fetchText(
+  "https://en.wikipedia.org/wiki/OpenAI",
+);
+const $ = load(html);
+// Cheerio accepts a CSS selector, here we pick the second <p>
+const intro = $("p:nth-of-type(2)").first().text();
+console.log(intro);
+```
+
```
Comment (stevekrouse) [inline src/content/docs/guides/web-scraping.mdx:62]:
```suggestion
Outputs:
```

File: src/content/docs/integrations/Databases/upstash.mdx [hunk 15]
```diff
@@ -50,17 +50,13 @@ If you set it up correctly, you should be able to copy & paste this Val and have
...
-export const upstashEx = (async () => {
-  const { Redis } = await import("npm:@upstash/redis");
-  const redis = new Redis({
-    url: process.env.upstashURL,
-    token: process.env.upstashToken,
-  });
-  console.log(await redis.set("foo", "bar"));
-  console.log(await redis.get("foo"));
-})();
+const { Redis } = await import("npm:@upstash/redis");
```
Comment (stevekrouse) [inline src/content/docs/integrations/Databases/upstash.mdx:53]: make static import

File: src/content/docs/integrations/Databases/upstash.mdx [hunk 16]
```diff
@@ -69,17 +65,13 @@ JSON is automatically stringified and parsed so you can set it and get it
...
-export const upstashJSONEx = (async () => {
-  const { Redis } = await import("npm:@upstash/redis");
-  const redis = new Redis({
-    url: process.env.upstashURL,
-    token: process.env.upstashToken,
-  });
-  await redis.set("json-ex-1", { a: { b: "nested json" } });
-  return ((await redis.get("json-ex-1")) as any).a.b;
-})();
+const { Redis } = await import("npm:@upstash/redis");
```
Comment (stevekrouse) [inline src/content/docs/integrations/Databases/upstash.mdx:68]: make static import

File: src/content/docs/integrations/GitHub/get-a-github-user.mdx [hunk 17]
```diff
@@ -3,22 +3,45 @@ title: Get a Github user
...
-<Val url="https://www.val.town/embed/vtdocs.getSteve" />
+export const getSteve = getGithubUser("stevekrouse");
+```
 
 You can also do this with [octokit.js](https://github.com/octokit/octokit.js). Reminder: import npm packages by prepending the package name with `npm:`.
 
-<Val url="https://www.val.town/embed/vtdocs.getGithubUserViaOctokit" />
+```ts title="@vtdocs/getGithubUserViaOctokit" val
+export const getGithubUserViaOctokit = async (username: string) => {
+  const { Octokit } = await import("npm:@octokit/core");
```
Comment (stevekrouse) [inline src/content/docs/integrations/GitHub/get-a-github-user.mdx:37]: convert to static import

File: src/content/docs/integrations/Google Sheets.mdx [hunk 18]
```diff
@@ -127,6 +125,27 @@ It requires 4 arguments:
...
+
+// Reading from the top of the sheet
+const data = await gsheet_call(
+  Deno.env.get("google_sa"),
+  "1LDgOhO6Fxg2wt5rGFH29t6XYbmKe6fXI7fLSFaqZkDA",
+  "GET",
+  "values/A1:C1?majorDimension=ROWS",
+  {},
+);
+export const gsheet_append_example = data["values"][0];
```
Comment (stevekrouse) [inline src/content/docs/integrations/Google Sheets.mdx:148]:
```suggestion
console.log(data["values"][0])
```

File: src/content/docs/integrations/airtable.mdx [hunk 19]
```diff
@@ -60,21 +58,17 @@ the value, then click **Add**.
...
-    baseId: "appXSrKDlwbAijRmD",
-    tableName: "All content",
-  });
-  // Sample data from: https://blog.airtable.com/database-vs-spreadsheet/
-  const results = await airtable.select();
-  return results;
-})();
+const { Airtable } = await import(
+  "https://deno.land/x/airtable@v1.1.1/mod.ts"
+);
```
Comment (stevekrouse) [inline src/content/docs/integrations/airtable.mdx:63]: convert to static import

File: src/content/docs/integrations/airtable.mdx [hunk 20]
```diff
@@ -60,21 +58,17 @@ the value, then click **Add**.
...
+const airtable = new Airtable({
+  apiKey: Deno.env.get("airtable_pat"),
+  baseId: "appXSrKDlwbAijRmD",
+  tableName: "All content",
+});
+// Sample data from: https://blog.airtable.com/database-vs-spreadsheet/
+const results = await airtable.select();
+export const airtable_deno_sample = results;
```
Comment (stevekrouse) [inline src/content/docs/integrations/airtable.mdx:71]:
```suggestion
console.log(results)
```

File: src/content/docs/integrations/telegram.mdx [hunk 21]
```diff
@@ -50,26 +48,47 @@ When Telegram users send messages to your bot, they will be forwarded to your
...
 
 ### Tell your bot about the webhook handler
 
 Use `@vtdocs.telegramSetWebhook` to tell your bot where to send webhooks.
 
 ```ts val
-import process from "node:process";
 import { telegramSetWebhook } from "https://esm.town/v/vtdocs/telegramSetWebhook?v=2";
 
+const token = Deno.env.get("telegramBotToken");
```
Comment (stevekrouse) [inline src/content/docs/integrations/telegram.mdx:83]:
```suggestion
```

File: src/content/docs/integrations/telegram.mdx [hunk 22]
```diff
@@ -50,26 +48,47 @@ When Telegram users send messages to your bot, they will be forwarded to your
...
 export const setWebhookExample = telegramSetWebhook(
-  process.env.telegramBotToken,
+  token,
```
Comment (stevekrouse) [inline src/content/docs/integrations/telegram.mdx:85]:
```suggestion
```

File: src/content/docs/integrations/telegram.mdx [hunk 23]
```diff
@@ -50,26 +48,47 @@ When Telegram users send messages to your bot, they will be forwarded to your
...
   {
     // Replace this with _your_ web endpoint
     url: "https://neverstew-telegramWebhookEchoMessage.web.val.run",
     // Optionally, filter what kind of updates you want to receive here
     allowed_updates: ["message"],
-    secret_token: process.env.telegramWebhookSecret,
+    secret_token: token,
```
Comment (stevekrouse) [inline src/content/docs/integrations/telegram.mdx:91]:
```suggestion
    secret_token: Deno.env.get("telegramBotToken"),
```

File: src/content/docs/types/http/routing.mdx [hunk 24]
```diff
@@ -3,30 +3,63 @@ title: Routing
...
+```
 
 ### Peko
 
 And one with [Peko](https://peko.deno.dev/):
 
-<Val url="https://www.val.town/embed/tmcw.pekoExample" />
+```ts val
+export const pekoExample = async (request) => {
+  const Peko = await import("https://deno.land/x/peko@2.0.0/mod.ts");
```
Comment (stevekrouse) [inline src/content/docs/types/http/routing.mdx:27]: Covert to static import

File: src/content/docs/types/http/routing.mdx [hunk 25]
```diff
@@ -3,30 +3,63 @@ title: Routing
...
+```
 
 ### nhttp
 
 And [nhttp](https://github.com/nhttp/nhttp):
 
-<Val url="https://www.val.town/embed/tmcw.nhttpExample" />
+```ts val
+export const nhttpExample = async (request) => {
+  const { nhttp } = await import("npm:nhttp-land@1");
```
Comment (stevekrouse) [inline src/content/docs/types/http/routing.mdx:41]: static import

File: src/content/docs/types/http/routing.mdx [hunk 26]
```diff
@@ -3,30 +3,63 @@ title: Routing
...
 
 ### itty-router
 
 A super tiny example with [itty-router](https://itty.dev/itty-router):
 
-<Val url="https://www.val.town/embed/tmcw.ittyRouterExample" />
+
+```ts val
+export const ittyRouterExample = async (request: Request) => {
+  const { Router, json } = await import("npm:itty-router@4");
```
Comment (stevekrouse) [inline src/content/docs/types/http/routing.mdx:60]: static import

**Review by stevekrouse — CHANGES_REQUESTED** [review]
Great work! A asked for a couple little nits, but looks good to go!

File: src/content/docs/guides/web-scraping.mdx (hunk 14 above)
Comment (Xkonti) [reply src/content/docs/guides/web-scraping.mdx:62]: Hmm... `Outputs` is very generic. I would recommend on settling on 2 very distinct words:
- `Logs` - results of `console.log` which correspond to the `Logs` tab in the interface
- `Exports` - results of exporting a value from val

The code block below the val has a title `Logs` to indicate the output generated by `console.log`. It looks like this:
image

**Comment by Xkonti** [comment]
@stevekrouse All but one requested changes implemented. Please review the conversation about adding `Output`.

File: src/content/docs/guides/web-scraping.mdx (hunk 14 above)
Comment (stevekrouse) [reply src/content/docs/guides/web-scraping.mdx:62]: Yeah that's perfect, ignore my comment then

**Pull request merged**